docs/trestle_author.md

```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
+  For YAML header we validate the structure and the presence of the "keys" and not the "values".
 
-```markdown
-# Template heading 1
+  The document YAML header(metadata) is said to be **_valid_** against the template, if and only if:
 
-# Template heading 2
-## Template sub heading
-```
+  1. It contains **all** keys from the template EXCEPT the keys that are either:
+     1. Start with `x-trestle`
+     1. Are listed under the `x-trestle-ignore` key (i.e. `x-trestle-ignore: not-important-field, bought-in`)
+  1. The version under `x-trestle-template-version: ` key is matching the template.
+  1. No new keys were added to the document.
 
-The following document is acceptable:
-
-```markdown
-# Template heading 1
-Content for heading one
-## Non-required sub header
-Content for non-required sub header
-# Template heading 2
-Content for heading two
-## Template sub heading
-Content for template sub heading
-### non required sub-sub heading
-This sub-sub heading is okay
-```
+  For example, this change to the YAML header above is acceptable:
 
-However, violations such as adding or removing a heading at a level that has been templated is not acceptable e.g.:
-
-```markdown
-# Template heading 1
-Content for heading one
-## Non-required sub header
-Content for non-required sub header
-# Template heading 2
-Content for heading two
-## Template sub heading
-Content for template sub heading
-## sub heading that violates template
-This sub heading is NOT okay
-# Top level heading that is not okay
-```
+  ```markdown
+  ---
+  groceries:
+      grocery1:
+          name: pear
+          type: fruit
+          bought-in: Superstore
+      grocery2:
+          name: potato
+          type: vegetable
+          bought-in: Local market
+  date: 10-10-2023
+  not-important-field: not important
+  x-trestle-version: 1.0.0
+  x-trestle-ignore: not-important-field, bought-in
+  ---
+  # The rest of my document
+  ```
 
-For each of the headings - the text of the heading is enforced with one caveat:
+  ### Enforcing heading structure
 
-- If the template heading text is wrapped in curly brackets `{}` then the name is not measured e.g. `# {Insert title here}`.
+  Markdown headings are considered to be nested based on the heading level (i.e. number of `#`). For example, `## 1.1 Lower level heading` is below `# 1. top level heading`.
 
-### Strict header / heading conformance mechanisms
+  The document body is said to be **_valid_** against the template, if and only if,
 
-Two mechanisms are provided to enforce metadata within markdown documents. The first is the yaml header, as used by technologies such as jekyll, the second is a markdown 'governed heading\` where templating of the content is enforced.
-Use of the yaml header is strongly encouraged as a first preference.
+  1. It contains **_all_** the headings provided in the template.
+  1. No new headings were added at the top level (i.e. `# New heading` is not allowed).
```

Comment:
Can a heading be repeated in a template?  This isn't so much a question for the docs as an actual question about how it works.  If there are any issues with duplicate headings it should probably be mentioned.

Reply:
With the latest fix for duplicate headers this is not a problem anymore. If there is a duplicate, both of them will be validated